include/z64cutscene.h

```diff
@@ -3,59 +3,429 @@
+#define CS_CMD_CAM_AT CS_CMD_CAM_AT_POINTS
+#define CS_CMD_SET_LIGHTING CS_CMD_LIGHT_SETTING
+#define CS_CMD_CAM_EYE_REL_TO_PLAYER CS_CMD_CAM_EYE_POINTS_REL_TO_PLAYER
+#define CS_CMD_CAM_AT_REL_TO_PLAYER CS_CMD_CAM_AT_POINTS_REL_TO_PLAYER
+#define CS_CMD_09 CS_CMD_RUMBLE_CONTROLLER
+#define CS_CMD_SET_PLAYER_ACTION CS_CMD_PLAYER_CUE
+#define CS_CMD_SCENE_TRANS_FX CS_CMD_TRANSITION
+#define CS_CMD_TEXTBOX CS_CMD_TEXT
+#define CS_CMD_PLAYBGM CS_CMD_PLAY_SEQ
+#define CS_CMD_STOPBGM CS_CMD_STOP_SEQ
+#define CS_CMD_FADEBGM CS_CMD_FADE_SEQ
+#define CS_CMD_SETTIME CS_CMD_TIME
+#define CS_CMD_SET_ACTOR_ACTION_1 CS_CMD_ACTOR_CUE_0_0
+#define CS_CMD_SET_ACTOR_ACTION_2 CS_CMD_ACTOR_CUE_1_0
+#define CS_CMD_SET_ACTOR_ACTION_4 CS_CMD_ACTOR_CUE_2_0
+#define CS_CMD_SET_ACTOR_ACTION_3 CS_CMD_ACTOR_CUE_3_0
+#define CS_CMD_SET_ACTOR_ACTION_5 CS_CMD_ACTOR_CUE_4_0
+#define CS_CMD_SET_ACTOR_ACTION_6 CS_CMD_ACTOR_CUE_5_0
+#define CS_CMD_SET_ACTOR_ACTION_7 CS_CMD_ACTOR_CUE_6_0
+#define CS_CMD_SET_ACTOR_ACTION_8 CS_CMD_ACTOR_CUE_7_0
+#define CS_CMD_SET_ACTOR_ACTION_9 CS_CMD_ACTOR_CUE_8_0
+#define CS_CMD_SET_ACTOR_ACTION_10 CS_CMD_ACTOR_CUE_9_0
+#define CS_CMD_TERMINATOR CS_CMD_DESTINATION
+
+typedef enum {
+    /* 0x01 */ CS_MISC_RAIN = 1,
+    /* 0x02 */ CS_MISC_LIGHTNING,
+    /* 0x03 */ CS_MISC_SET_FLAG_0, // also sets flag 2 if in Temple of Time
+    /* 0x04 */ CS_MISC_UNIMPLEMENTED_4,
+    /* 0x06 */ CS_MISC_LIFT_FOG = 6,
+    /* 0x07 */ CS_MISC_CLOUDY_SKY,
+    /* 0x08 */ CS_MISC_FADE_KOKIRI_GRASS_ENV_ALPHA,
+    /* 0x09 */ CS_MISC_SNOW,
+    /* 0x0A */ CS_MISC_SET_FLAG_1,
+    /* 0x0B */ CS_MISC_DEKU_TREE_DEATH,
+    /* 0x0C */ CS_MISC_STOP,
+    /* 0x0D */ CS_MISC_TRIFORCE_FLASH,
+    /* 0x0E */ CS_MISC_SET_LOCKED_VIEWPOINT,
+    /* 0x0F */ CS_MISC_SHOW_TITLE_CARD,
+    /* 0x10 */ CS_MISC_QUAKE_START,
+    /* 0x11 */ CS_MISC_QUAKE_STOP,
+    /* 0x12 */ CS_MISC_STOP_STORM_AND_ADVANCE_TO_DAY,
+    /* 0x13 */ CS_MISC_SET_FLAG_FAST_WINDMILL,
+    /* 0x14 */ CS_MISC_SET_FLAG_WELL_DRAINED,
+    /* 0x15 */ CS_MISC_SET_FLAG_LAKE_HYLIA_RESTORED,
+    /* 0x16 */ CS_MISC_VISMONO_BLACK_AND_WHITE,
+    /* 0x17 */ CS_MISC_VISMONO_SEPIA,
+    /* 0x18 */ CS_MISC_HIDE_ROOM,
+    /* 0x19 */ CS_MISC_TIME_ADVANCE_TO_NIGHT,
+    /* 0x1A */ CS_MISC_SET_TIME_BASED_LIGHT_SETTING,
+    /* 0x1B */ CS_MISC_RED_PULSATING_LIGHTS,
+    /* 0x1C */ CS_MISC_HALT_ALL_ACTORS,
+    /* 0x1D */ CS_MISC_RESUME_ALL_ACTORS,
+    /* 0x1E */ CS_MISC_SET_FLAG_3,
+    /* 0x1F */ CS_MISC_SET_FLAG_4,
+    /* 0x20 */ CS_MISC_SANDSTORM,
+    /* 0x21 */ CS_MISC_SUNSSONG_START,
+    /* 0x22 */ CS_MISC_FREEZE_TIME,
+    /* 0x23 */ CS_MISC_LONG_SCARECROW_SONG
+} CutsceneMiscType;
+
+typedef enum {
+    /* 0x00 */ CS_TEXT_NORMAL,
+    /* 0x01 */ CS_TEXT_CHOICE,
+    /* 0x02 */ CS_TEXT_OCARINA_ACTION,
+    /* 0x03 */ CS_TEXT_GORON_RUBY, // use `altTextId1` if sapphire is already obtained
+    /* 0x04 */ CS_TEXT_ZORA_SAPPHIRE // use `altTextId1` if ruby is already obtained
```

Comment:
Are the comments/names swapped?

Reply:
they arent. the point is that if you are for example, currently getting goron ruby, and you already have the sapphire, alternate text can be shown.
this is a left over from a time when you could complete these dungeons in any order. its not relevant for the final game. i believe the mechanism it itself still works though and the alternate text will show

Reply:
Oh ok I read cs_text_zora_sapphire as "conditioned by having sapphire" instead of "used for the zora sapphire cutscene"
I expected it to be named by behavior and not usage

Also I think you can watch the sapphire cutscene before the ruby cutscene, I'm super rusty on how vanilla works but if bombs is the only thing preventing going to jabu you can just get them and go beat jabu before dc?

Reply:
Why not keep them as is, but to prevent confusion maybe have a  "used by sapphire/ruby cutscene" comment?